symbolic-demangle/vendor/swift/1-arguments.patch

```diff
@@ -1,45 +1,135 @@
-commit 43fca7dd2617ac93f338b5257a2e57c43dcb8154
-Author: Sebastian Zivota <loewenheim@mailbox.org>
-Date:   Thu Dec 2 16:15:35 2021 +0100
+commit eaae91fd522031a2aaafed367414f942f1c83d22
+Author: fantexi023 <1249434307@qq.com>
+Date:   Thu Jul 18 17:04:45 2024 +0800
 
-    Apply patch
+    pass cargo build
 
-diff --git a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
-index db32dbd..f48e1c2 100644
---- a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
-+++ b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
-@@ -59,6 +59,7 @@ struct DemangleOptions {
-   bool ShortenArchetype = false;
-   bool ShowPrivateDiscriminators = true;
-   bool ShowFunctionArgumentTypes = true;
-+  bool ShowFunctionReturnType = true;
```

Comment:
You mentioned this in the changed `swiftdemangle.cpp` file. This option is indeed not part of upstream Swift, but we patched that in via this small patch here.

Reply:
Thank you for your reply. 
I also saw that the CI process gave five errors, among which the problem of [Rust Test on macos-latest ](https://github.com/getsentry/symbolic/actions/runs/9989076444/job/27607390557#logs)should be that the SWIFT_STDLIB_HAS_TYPE_PRINTING did not take effect, which may be caused by the passing exception of CMakeLists.txt. I cancelled the restriction of SWIFT_STDLIB_HAS_TYPE_PRINTING this macro locally. The other four problems correspond to the same error, but I did not encounter this error on my own Mac.
Lastly, regarding ShowFunctionReturnType I still don't understand, how do you work through patches? Isn't the final demangle still implemented through the official Swift interface?

Reply:
> You mentioned this in the changed `swiftdemangle.cpp` file. This option is indeed not part of upstream Swift, but we patched that in via this small patch here.

yeah, I do understand your explanation for ShowFunctionReturnType. TXS!